src/Raven.Client/Json/Converters/LazyStringValueJsonConverter.cs

```diff
@@ -1,37 +1,36 @@
 ﻿using System;
+using System.Runtime.Serialization;
 using Newtonsoft.Json;
 using Sparrow.Json;
 
 namespace Raven.Client.Json.Converters
 {
     internal sealed class LazyStringValueJsonConverter : RavenJsonConverter
     {
-        //Todo To consider put context as part of the converter and not part of the reader
-        //public BlittableJsonConverter(JsonOperationContext context)
-        //{
-        //    _context = context;
-        //}
-
-
         public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
         {
             writer.WriteValue(value);
         }
 
         public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
         {
-            //Todo To consider put context as part of the converter and not part of the reader
-            //if (reader.Value is LazyStringValue lazyString)
-            //{
-            //    return lazyString.Clone(_context);
-            //}
-            //throw new Exception($"Value must be {nameof(BlittableJsonReaderObject)}");
+            var blittableReader = reader as BlittableJsonReader;
+            if (blittableReader == null)
+            {
+                throw new Exception($"{nameof(BlittableJsonReader)} must to be used for convert to {nameof(BlittableJsonReaderObject)}");
+            }
+
+            if (blittableReader.Value == null)
+            {
+                return null;
+            }
 
-            if (reader is BlittableJsonReader blittableReader)
+            //Todo It will be better to change the reader to set the value as LazyStringValue 
+            if (blittableReader.Value is string strValue)
             {
-                return blittableReader.ReadAsLazyStringValue();
+                return blittableReader.Context.GetLazyString(strValue);
```

Comment:
General note: here we see that using the Newtonsoft infrastructure forces us to do a string materialization, only to create a LazyStringValue out of it once again, when we create the object instance. We need to understand whether we really want to continue using Newtonsoft.Json for the transaction merging..

Reply:
We can probably do this using `DynamicJsonValue` and directly serialization?

Reply:
yes